### ISSUE TITLE
Inconsistent displays of Project Conflicts

### ISSUE THREAD

**Issue by znmeb** [hackoregon/transportation-frontend2#7]
Sometimes we see pushpins, sometimes we see lines ... sometimes both ... seems to vary by browser., etc.

**Comment by sanjuroj** [comment]
And just to be clear, I believe the data is all there, but the markers aren't displaying properly.  

Are you not seeing lines where you think there should be lines?

**Comment by znmeb** [comment]
I see both pushpins and lines here ... is that the way it should be?

**Comment by znmeb** [comment]
I had to post the JSON as a Gist - which GitHub lovingly renders!!

https://gist.github.com/znmeb/bd423c7d33634774f8ac8166bfa2a094

**Comment by sanjuroj** [comment]
Yup, there should be both.  It's potentially displaying points, lines, and polygons.  Any feature that is close to any other feature, regardless of whether the types match, should be displayed.

**Comment by znmeb** [comment]
So it's working here? Firefox Nightly (55.0a1 (2017-05-07) (64-bit)) on Linux host

**Comment by sanjuroj** [comment]
Which "here" do you mean?  I can see the pins on chrome and firefox (and safari) but others can't.

**Comment by znmeb** [comment]
"Here" being my workstation at home

**Comment by sanjuroj** [comment]
Mine too.  Others not so much.  Not sure why.

**Comment by znmeb** [comment]
Has anyone reported an issue besides people at the table Saturday?

**Comment by sanjuroj** [comment]
My nephew tried it yesterday from my house and no markers for him.  Lee tried it at home (I think) and it worked on Chrome, the whole app failed on IE, and no icons on mobile.

**Comment by znmeb** [comment]
![jhmp-5u8](https://cloud.githubusercontent.com/assets/4938/26080937/88edf06c-397d-11e7-9917-68710bd111bb.jpg)
I have screen shots from Android and they're bad ... Posting them in a bit

This is Firefox Nightly on Android

**Comment by znmeb** [comment]
And this is the default Google built-in browser:


![l3s9siah](https://cloud.githubusercontent.com/assets/4938/26081094/2b887144-397e-11e7-828f-db372c517326.jpg)

**Comment by znmeb** [comment]
We deliberately did not do mobile first / responsive as an "executive decision" so I'm not sure these are relevant.

**Comment by znmeb** [comment]
I'm going to test this on Microsoft browser test virtual machines (https://developer.microsoft.com/en-us/microsoft-edge/tools/vms/). If I can't reproduce it I think we should close this. If anyone else is seeing rendering issues, let me know and I'll test theirs too.

**Comment by znmeb** [comment]
IE 8 - site does not display at all, has an error. To be fair, *GitHub* no longer supports IE8!
Webpage error details
User Agent: Mozilla/4.0 (compatible; MSIE 8.0; Windows NT 6.1; Trident/4.0; SLCC2; .NET CLR 2.0.50727; .NET CLR 3.5.30729; .NET CLR 3.0.30729; Media Center PC 6.0; .NET4.0C)
Timestamp: Wed, 17 May 2017 06:44:05 UTC
Message: Expected identifier, string or number
Line: 1
Char: 547
Code: 0
URI: http://transportation.civicpdx.org/main.7be655c58b6f0a216b91.js

**Comment by znmeb** [comment]
Windows 10 / MS Edge: works - have both pushpins and lines, just like Firefox and Chromium on Linux.

**Comment by znmeb** [comment]
Windows 7 32-bit: Firefox works, IE 11 does not. `civicpdx.org` displays fine, but when you click on any of the `Collection` links you're taken to the right page but it's totally blank. Can someone else with Windows and IE 11 confirm this? I'm running in a VirtualBox VM and want to rule that out as the problem.

**Comment by sanjuroj** [comment]
All I have is Windows 10

**Comment by znmeb** [comment]
You should have Edge (which works) and you should be able to install IE 11 concurrently with Edge. I don't see any point in supporting older than 11, but there will be a lot of people out there with 11.

**Comment by znmeb** [comment]
It turns out on 64-bit Windows 10 your only Microsoft option is Edge - IE 11 is only supported on Win 7 - 8.1, and on Win 10 32-bit.

Given how aggressive Microsoft is with updates, we only need to worry about IE 11 and Edge, and Edge works. Should I put out a call in Slack for people with IE 11 to test on a real machine? I don't trust these virtual machines, even though I've updated them with all the Microsoft updates to date (which takes a couple of hours in a VM.)

**Comment by sanjuroj** [comment]
I should be able to test on Windows 10 soon...

**Comment by znmeb** [comment]
It'll work - it does on my Windows 10 VM and on my laptop in both Firefox and Chrome.

**Comment by sanjuroj** [comment]
On my Windows 10 desktop, Edge shows markers, Chrome doesn't.

**Comment by znmeb** [comment]
Crap! Let me put Chrome in my virtual machine and see what's up!

**Comment by znmeb** [comment]
All three options work on Edge, Chrome and Firefox in the virtual machine. It is Windows 10 x64, Preview 15.15063

**Comment by znmeb** [comment]
I'll test Chrome and Firefox on Windows 8.1 and Windows 7, given that IE 11 doesn't work on either of them. I hate having to put one of those "Best with modern browsers" warnings on a site - that's so 1995. :-(  We should probably ask Davide if he knows of any React IE 11 incompatibilities.

**Comment by znmeb** [comment]
OK ... Firefox and Chrome both work on Windows 8.1.

Symptom review:

- Everything works on Windows 10 64 bit 15.15053 in Edge, Chrome and Firefox
- Everything works on Windows 8.1 32 bit in Chrome and Firefox
- On Windows 8.1, Internet Explorer 11 displays <civicpdx.org> correctly but *all* of the links to the collections display as a blank page, including <transportation.civicpdx.org>.

**Comment by znmeb** [comment]
https://github.com/facebook/react/issues/8379 

It looks like load order of Javascript libraries matters ... a long thread on the issue there but it looks like it may be what we have.

**Comment by sanjuroj** [comment]
Hmmm, not sure if that post is quite what we are seeing.  I suspect that the icons are not being pulled in from the component library npm module.  The React aspects that are discussed in that post seem to be working ok for us.

**Comment by znmeb** [comment]
we need to separate the two issues - I should have filed the IE 11 thing as a new one.

I have not seen anything in my testing like missing icons - just the blank pages on IE 11.

**Comment by znmeb** [comment]
OK ... I filed issue #10  but I'm going to move it to the platform and close it here

**Comment by znmeb** [comment]
Closed and refiled at https://github.com/hackoregon/civic-platform/issues/10
